server/src/main/java/org/jboss/as/server/parsing/ExtensionHandler.java

```diff
@@ -0,0 +1,23 @@
+package org.jboss.as.server.parsing;
+
+import org.jboss.as.controller.parsing.Namespace;
+import org.jboss.as.controller.parsing.ProfileParsingCompletionHandler;
+import org.jboss.dmr.ModelNode;
+import org.jboss.staxmapper.XMLExtendedStreamReader;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+import javax.xml.stream.XMLStreamException;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Heiko Braun
+ * @since 27/11/15
+ */
+public interface ExtensionHandler {
+    void parseExtensions(XMLExtendedStreamReader reader, ModelNode address, Namespace namespace, List<ModelNode> list) throws XMLStreamException;
+
+    Set<ProfileParsingCompletionHandler> getParsingCompletionHandler();
```

Comment:
Use getProfileParsingCompletionHandlers() to be consistent with ExtensionRegistry. Plus you're missing an 's'. ;)